concent_api/core/tests/utils.py

```diff
@@ -204,6 +205,16 @@ def _test_database_objects(
             self.assertEqual(ReceiveStatus.objects.last().delivered,        receive_delivered_status)
             self.assertEqual(ReceiveStatus.objects.last().message.task_id,  task_id)
 
+    def _test_subtask_state(self, task_id, subtask_id, subtask_state, provider_key, requestor_key, next_deadline = None):
+        last_subtask = Subtask.objects.last()
+        self.assertEqual(last_subtask.task_id,              task_id)
+        self.assertEqual(last_subtask.subtask_id,           subtask_id)
+        self.assertEqual(last_subtask.state,                subtask_state)
+        self.assertEqual(last_subtask.provider.public_key,  provider_key)
+        self.assertEqual(last_subtask.requestor.public_key, requestor_key)
+        if next_deadline is not None:
```

Comment:
For passive states `next_deadline` is `None`, and we would like to check if it is `None`. For active states we want to check it exactly because we know all the values. So I think this `if` is not needed here.

Reply:
I think that some sort of `if` is needed. When `next_deadline` is `None`, I'm not going to be able to `assertEqual` `timestamp()` from it. Maybe it's better to just check if Subtask is in passive or active state, and base on that assert `timestamp()` or `None`?

Reply:
Then we can skip situation when in passive state there will be other value then `None`. What you can do is check if `last_subtask.next_deadline` is None, or not, if it is assign to some variable, if it isn't assign it to some variable but as timestamp, then compare this variable in .assertEqual to next_deadline.